### PR TITLE
ZBUG-2397: Added cookie SameSite=Strict and zmlocalconfig attribute

### PR DESCRIPTION
**Pull request by zimsuchitgupta**
Supported SameSite=Strict cookie in ZM_AUTH_TOKEN cookie.
For this added a localconfig varible in LC.java named zimbra_same_site_cookie
By default value will be 'Strict' but user can change it to Lax, None and ""(empty) for existing behavior.

**Testing Instruction:**

**1** - On login in webclient, by default user can see the SameSite=Strict cookie for ZM_AUTH_TOKEN.
http://synscreen.buf.synacor.com/view/a5d6a2.png
**2** - User can control behavior using zmlocalconfig

zmlocalconfig -e zimbra_same_site_cookie="Strict"
User can provide value like Strict, Lax, None or "" (Empty)

**3** - To disable this behavior we can set value of SameSite=""
http://synscreen.buf.synacor.com/view/a7d8d2.png

### CODE REVIEW

File: common/src/java/com/zimbra/common/util/ZimbraCookie.java [hunk 1]
```diff
@@ -70,7 +70,14 @@ private static void addCookie(HttpServletResponse response, String name, String
         }
         ZimbraCookie.setAuthTokenCookieDomainPath(cookie, ZimbraCookie.PATH_ROOT);
 
-        cookie.setSecure(secure);
+        String cookieVal = LC.zimbra_same_site_cookie.value();
```
Comment (silentsakky) [inline common/src/java/com/zimbra/common/util/ZimbraCookie.java:73]: hope this is used in JWT authentication also, can we check it?

Comment (silentsakky) [reply common/src/java/com/zimbra/common/util/ZimbraCookie.java:73]: can we also add unittests for this change?

Comment (zimsuchitgupta) [reply common/src/java/com/zimbra/common/util/ZimbraCookie.java:73]: @silentsakky Above method is also called for JWT token also can see here https://github.com/Zimbra/zm-mailbox/blob/develop/store/src/java/com/zimbra/cs/account/ZimbraJWToken.java#L301 but need to verify by doing testing.

**Review by log2akshat — APPROVED** [review]

File: common/src/java/com/zimbra/common/util/ZimbraCookie.java (hunk 1 above)
Comment (zimsuchitgupta) [reply common/src/java/com/zimbra/common/util/ZimbraCookie.java:73]: Thanks @silentsakky  for you help verifying feature for JWT token, as you shared and I also verified that this is working for JWT token too. 
JWT Strict

**Pull request merged**